tls/datadog_checks/tls/tls.py

```diff
@@ -0,0 +1,348 @@
+# (C) Datadog, Inc. 2019
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import socket
+import ssl
+from datetime import datetime
+from os.path import expanduser, isdir
+
+import service_identity
+from cryptography.hazmat.backends import default_backend
+from cryptography.x509 import load_der_x509_certificate, load_pem_x509_certificate
+from six import text_type
+from six.moves.urllib.parse import urlparse
+
+from datadog_checks.base import AgentCheck, ConfigurationError, is_affirmative
+
+from .utils import closing, days_to_seconds, get_protocol_versions, is_ip_address, seconds_to_days
+
+# Python 3 only
+PROTOCOL_TLS_CLIENT = getattr(ssl, 'PROTOCOL_TLS_CLIENT', ssl.PROTOCOL_TLS)
+
+
+class TLSCheck(AgentCheck):
+    SERVICE_CHECK_CAN_CONNECT = 'tls.can_connect'
+    SERVICE_CHECK_VERSION = 'tls.version'
+    SERVICE_CHECK_VALIDATION = 'tls.cert_validation'
+    SERVICE_CHECK_EXPIRATION = 'tls.cert_expiration'
+
+    DEFAULT_EXPIRE_DAYS_WARNING = 14
+    DEFAULT_EXPIRE_DAYS_CRITICAL = 7
+    DEFAULT_EXPIRE_SECONDS_WARNING = days_to_seconds(DEFAULT_EXPIRE_DAYS_WARNING)
+    DEFAULT_EXPIRE_SECONDS_CRITICAL = days_to_seconds(DEFAULT_EXPIRE_DAYS_CRITICAL)
+
+    def __init__(self, name, init_config, instances):
+        super(TLSCheck, self).__init__(name, init_config, instances)
+
+        self._name = self.instance.get('name')
+        self._local_cert_path = self.instance.get('local_cert_path', '')
+        self._timeout = float(self.instance.get('timeout', 10))
+
+        server = self.instance.get('server', '')
+        parsed_uri = urlparse(server)
+
+        # Handle IP addresses, see: https://bugs.python.org/issue754016
+        if not parsed_uri.hostname:
+            parsed_uri = urlparse('//{}'.format(server))
+
+        self._server = parsed_uri.hostname
+
+        # TODO: Support (implement) UDP
+        # https://chris-wood.github.io/2016/05/06/OpenSSL-DTLS.html
+        transport = self.instance.get('transport', 'tcp').lower()
+        if transport == 'udp':
+            # SOCK_DGRAM
```

Comment:
Did you write that as it is what it should be?

Reply:
Yup!